integration-tests/src/test/java/oracle/kubernetes/operator/ItMonitoringExporter.java

```diff
@@ -1162,69 +1254,159 @@ private static void installWebHookAndAlertManager() throws Exception {
+
+      getWebHook.append("cd " + webhookDir)
+              .append(" && ")
+              .append(" cp webhook-linux-amd64/webhook .");
+
+      TestUtils.exec(getWebHook.toString());
+
+    }
+    logger.info("building webhook image with scaling");
+
+    String crdCmd = "cd " + webhookDir + " && docker build . -t webhook:latest";
+    TestUtils.exec(crdCmd);
+
+     String internalOpCert = getInternalOpCert(operator);
+     replaceStringInFile(webhookDir+ "/webhook-deployment.yaml","@INTERNAL_OPERATOR_CERT@", internalOpCert);
+    // install webhook
+    logger.info("installing webhook ");
+
+    crdCmd = "kubectl apply -f " + webhookDir + "/webhook-deployment.yaml";
+    TestUtils.exec(crdCmd);
+    crdCmd = "kubectl apply -f " + webhookDir + "/crossrbac_monitoring.yaml";
+    TestUtils.exec(crdCmd);
+    String webhookPod = getPodName("name=webhook", "monitoring");
+    TestUtils.checkPodReady(webhookPod, "monitoring");
+
+  }
+
+  /**
+   * Uninstall Prometheus and Grafana using helm chart
    *
    * @throws Exception if could not run the command successfully to uninstall deployments
    */
   private static void uninstallWebHookPrometheusGrafanaViaChart() throws Exception {
+    String crdCmd;
+    String podName;
     logger.info("Uninstalling webhook");
-    String crdCmd = "kubectl delete -f " + monitoringExporterEndToEndDir + "/webhook/server.yaml";
-    ExecCommand.exec(crdCmd);
-    Thread.sleep(15000);
-    crdCmd = "kubectl delete ns webhook ";
-    ExecCommand.exec(crdCmd);
-    logger.info("Uninstalling grafana");
-    // uninstall grafana
-    crdCmd = "helm delete --purge grafana";
-    ExecCommand.exec(crdCmd);
-    crdCmd = "kubectl -n monitoring delete secret grafana-secret";
-    ExecCommand.exec(crdCmd);
-    crdCmd = "kubectl delete -f " + monitoringExporterEndToEndDir + "/grafana/persistence.yaml";
-    ExecCommand.exec(crdCmd);
-    Thread.sleep(15000);
+    try {
+      podName = getPodName("app=webhook", "webhook");
+      crdCmd = "kubectl delete -f " + monitoringExporterEndToEndDir + "/webhook/server.yaml";
+      ExecCommand.exec(crdCmd);
+      TestUtils.checkPodDeleted(podName, "webhook");
+      crdCmd = "kubectl delete ns webhook ";
+      ExecCommand.exec(crdCmd);
+    } catch (AssertionError assertionError) {
+      // ignore, pod may not be created
+    }
 
-    // uninstall prometheus
-    logger.info("Uninstalling prometheus");
-    crdCmd = "helm delete --purge prometheus";
-    ExecCommand.exec(crdCmd);
-    Thread.sleep(15000);
-    logger.info("Uninstalling prometheus persistence ");
-    crdCmd = "kubectl delete -f " + monitoringExporterEndToEndDir + "/prometheus/persistence.yaml";
-    ExecCommand.exec(crdCmd);
-    Thread.sleep(15000);
+    try {
+      podName = getPodName("app=grafana", "monitoring");
+      logger.info("Uninstalling grafana");
+      // uninstall grafana
+      crdCmd = "helm delete --purge grafana";
+      ExecCommand.exec(crdCmd);
+      Thread.sleep(15000);
```

Comment:
15 seconds sleep at multiple places in the test code, it increases the test run time. Can you see if 15 seconds is really needed and if it can be checked in loop instead of hard sleep.